ci/cudf_pandas_scripts/run_tests.sh

```diff
@@ -36,8 +36,10 @@ if [ "$no_cudf" = true ]; then
     echo "Skipping cudf install"
 else
     RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
-    RAPIDS_PY_WHEEL_NAME="cudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 ./local-cudf-dep
-    python -m pip install $(ls ./local-cudf-dep/cudf*.whl)[test,cudf-pandas-tests]
+    RAPIDS_PY_WHEEL_NAME="libcudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 cpp ./local-cudf-dep
+    RAPIDS_PY_WHEEL_NAME="cudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 python ./local-cudf-dep
+    python -m pip install "$(echo ./local-cudf-dep/libcudf_${RAPIDS_PY_CUDA_SUFFIX}*.whl)"
+    python -m pip install --find-links $(pwd)/local-cudf-dep "$(echo ./local-cudf-dep/cudf_${RAPIDS_PY_CUDA_SUFFIX}*.whl)[test,pandas-tests]"
```

Comment:
*(edit: moved to a threaded comment so all the debugging can be collapsed together when we figure it out)*

Still working through the `pandas` test failures, but I at least have been able to reproduce this locally. Ran the following on a machine with a couple V100s, running CUDA 12.2.

```shell
docker run \
    --rm \
    --gpus 1 \
    --env RAPIDS_BUILD_TYPE="pull-request" \
    --env RAPIDS_REF_NAME="pull-request/15483" \
    --env RAPIDS_REPOSITORY="rapidsai/cudf" \
    --env RAPIDS_SHA=42c5450 \
    -v $(pwd):/opt/work \
    -w /opt/work \
    -it rapidsai/citestwheel:cuda12.2.2-ubuntu22.04-py3.10 \
    bash

rm -rf ./local-cudf-dep
rm -rf ./test-results

bash ./ci/cudf_pandas_scripts/run_tests.sh
```

And saw exactly the test failures we're seeing in CI.

```text
FAILED python/cudf/cudf_pandas_tests/test_cudf_pandas.py::test_rmm_option_on_import[pool] - AssertionError: subprocess return code for env_value='pool' was non-zero (-8)
assert -8 == 0
 +  where -8 = CompletedProcess(args=['python', '-m', 'cudf.pandas', '/__w/cudf/cudf/python/cudf/cudf_pandas_tests/data/profile_basic.py'], returncode=-8, stdout='', stderr='').returncode
FAILED python/cudf/cudf_pandas_tests/test_cudf_pandas.py::test_rmm_option_on_import[managed_pool] - AssertionError: subprocess return code for env_value='managed_pool' was non-zero (-8)
assert -8 == 0
 +  where -8 = CompletedProcess(args=['python', '-m', 'cudf.pandas', '/__w/cudf/cudf/python/cudf/cudf_pandas_tests/data/profile_basic.py'], returncode=-8, stdout='', stderr='').returncode
FAILED python/cudf/cudf_pandas_tests/test_magics.py::test_magics_gpu - assert "Traceback (m...d 'IPython'\n" == ''
  + Traceback (most recent call last):
  +   File "/__w/cudf/cudf/python/cudf/cudf_pandas_tests/_magics_gpu_test.py", line 26, in <module>
  +     ipython_magics_gpu_test()
  +   File "/__w/cudf/cudf/python/cudf/cudf_pandas_tests/_magics_gpu_test.py", line 7, in ipython_magics_gpu_test
  +     from IPython.core.interactiveshell import InteractiveShell
  + ModuleNotFoundError: No module named 'IPython'
= 3 failed, 280 passed, 2 skipped, 4 xfailed, 15 warnings in 160.20s (0:02:40) =
```

([build link](https://github.com/rapidsai/cudf/actions/runs/9845343306/job/27181477763?pr=15483))

Reply:
Ok, narrowing this down further. Using the same `docker run` setup as above, just the following is enough to reproduce the issue.

```shell
# install wheels built in CI
RAPIDS_PY_WHEEL_NAME="libcudf_cu12" rapids-download-wheels-from-s3 cpp ./local-cudf-dep
RAPIDS_PY_WHEEL_NAME="cudf_cu12" rapids-download-wheels-from-s3 python ./local-cudf-dep
python -m pip install "$(echo ./local-cudf-dep/libcudf_cu12.whl)"
python -m pip install --find-links $(pwd)/local-cudf-dep "$(echo ./local-cudf-dep/cudf_cu12*.whl)[test,pandas-tests]"

# test exeuction of a tiny Python progam
cat > test.py <<EOF
import pandas as pd
EOF

CUDF_PANDAS_RMM_MODE=pool \
    python -m 'cudf.pandas' ./test.py
# Floating point exception (core dumped)

echo $?
# 136
```

That floating point exception happens for `CUDF_PANDAS_RMM_MODE` values `pool` and `managed_pool`. No issues show up when that's set to `async`, `cuda`, or `managed` (or unset).

```shell
CUDF_PANDAS_RMM_MODE=async \
    python -m 'cudf.pandas' ./test.py

echo $?
# 0
```

There's only one code path accessing the `CUDF_PANDAS_RMM_MODE` environment variable, in `cudf.pandas.install()`.

https://github.com/rapidsai/cudf/blob/cc8c86857df92801561d2fa3311d8da85895ff33/python/cudf/cudf/pandas/__init__.py#L28

https://github.com/rapidsai/cudf/blob/cc8c86857df92801561d2fa3311d8da85895ff33/python/cudf/cudf/pandas/__init__.py#L42-L48

Given that, the error is reproducible without `-m 'cudf.pandas'`, for the same set of RMM modes.

```shell
cat > ./test2.py <<EOF
import cudf.pandas
cudf.pandas.install()
EOF

CUDF_PANDAS_RMM_MODE=pool \
    python ./test2.py

# Floating point exception (core dumped)

echo $?
# 136
```

I'm not really sure where to go from there, could use some hints on what to investigate next.

Reply:
> ModuleNotFoundError: No module named 'IPython'

I see the root cause of this one. Typo in the name of an extra. Fixed in https://github.com/rapidsai/cudf/pull/15483/commits/9d05dfc5adddcd314f22549dfcc20c34eeb58e75.

Reply:
Thanks to @wence- 's suggestion, I was able to get more information by running this under `gdb`.

<details><summary>reproducible example (click me)</summary>

Note that this is using the `ci-conda` image... `gdb` was not happy with the `pyenv`-managed `python` in the `citestwheel` image.

```text
"/pyenv/shims/python": not in executable format: file format not recognized
```

Code to test:

```shell
docker run \
    --rm \
    --gpus 1 \
    --env RAPIDS_BUILD_TYPE="pull-request" \
    --env RAPIDS_REF_NAME="pull-request/15483" \
    --env RAPIDS_REPOSITORY="rapidsai/cudf" \
    --env RAPIDS_SHA=42c5450 \
    -v $(pwd):/opt/work \
    -w /opt/work \
    -it rapidsai/ci-conda:cuda12.2.2-ubuntu22.04-py3.10 \
    bash

conda install -c conda-forge gdb

RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
RAPIDS_PY_WHEEL_NAME="libcudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 cpp ./local-cudf-dep
RAPIDS_PY_WHEEL_NAME="cudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 python ./local-cudf-dep
python -m pip install "$(echo ./local-cudf-dep/libcudf_${RAPIDS_PY_CUDA_SUFFIX}*.whl)"
python -m pip install --find-links $(pwd)/local-cudf-dep "$(echo ./local-cudf-dep/cudf_${RAPIDS_PY_CUDA_SUFFIX}*.whl)[test,pandas-tests]"

cat > ./test2.py <<EOF
import cudf.pandas
cudf.pandas.install()
EOF

CUDF_PANDAS_RMM_MODE=pool gdb --args python ./test2.py
```

And then, within `gdb`

```text
(gdb) run
(gdb) bt
(gdb) py-bt
```

</details>

Looks like a conflict with `spdlog`.

```text
Thread 1 "python" received signal SIGFPE, Arithmetic exception.
0x00007fb5e0c51365 in void spdlog::pattern_formatter::handle_flag_<spdlog::details::scoped_padder>(char, spdlog::details::padding_info) ()
   from /opt/conda/lib/python3.10/site-packages/libcudf/lib64/libnvcomp.so
```

<details><summary>backtrace from 'bt' (click me)</summary>

```text
#0  0x00007fb5e0c51365 in void spdlog::pattern_formatter::handle_flag_<spdlog::details::scoped_padder>(char, spdlog::details::padding_info) ()
   from /opt/conda/lib/python3.10/site-packages/libcudf/lib64/libnvcomp.so
#1  0x00007fb5d6a61f59 in spdlog::pattern_formatter::compile_pattern_(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
   from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
#2  0x00007fb5d6a62c39 in spdlog::logger::set_pattern(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, spdlog::pattern_time_type) ()
   from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
#3  0x00007fb5d6a81483 in rmm::mr::pool_memory_resource<rmm::mr::device_memory_resource>::try_to_expand(unsigned long, unsigned long, rmm::cuda_stream_view) ()
   from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
#4  0x00007fb5d6a483c8 in ?? () from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
#5  0x0000561c3c2e7999 in type_call (kwds={'initial_pool_size': 27004082688}, args=(<rmm._lib.memory_resource.CudaMemoryResource at remote 0x7fb6b3cd7b10>,), type=0x7fb5d6b4c8e0)
    at /usr/local/src/conda/python-3.10.14/Objects/typeobject.c:1123
#6  _PyObject_MakeTpCall (tstate=0x561c3c5b30a0, callable=callable@entry=<type at remote 0x7fb5d6b4c8e0>, args=<optimized out>, nargs=1, keywords=keywords@entry=('initial_pool_size',))
    at /usr/local/src/conda/python-3.10.14/Objects/call.c:215
#7  0x0000561c3c2e3c53 in _PyObject_VectorcallTstate (kwnames=('initial_pool_size',), nargsf=<optimized out>, args=<optimized out>, callable=<type at remote 0x7fb5d6b4c8e0>, 
    tstate=<optimized out>) at /usr/local/src/conda/python-3.10.14/Include/cpython/abstract.h:112
#8  _PyObject_VectorcallTstate (kwnames=('initial_pool_size',), nargsf=<optimized out>, args=<optimized out>, callable=<type at remote 0x7fb5d6b4c8e0>, tstate=<optimized out>)
    at /usr/local/src/conda/python-3.10.14/Include/cpython/abstract.h:99
#9  PyObject_Vectorcall (kwnames=('initial_pool_size',), nargsf=<optimized out>, args=<optimized out>, callable=<type at remote 0x7fb5d6b4c8e0>)
    at /usr/local/src/conda/python-3.10.14/Include/cpython/abstract.h:123
#10 call_function (kwnames=('initial_pool_size',), oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7ffcbe3e2c10, tstate=<optimized out>)
    at /usr/local/src/conda/python-3.10.14/Python/ceval.c:5893
--Type <RET> for more, q to quit, c to continue without paging--
#11 _PyEval_EvalFrameDefault (tstate=<optimized out>, 
    f=Frame 0x7fb6b3d9e5c0, for file /opt/conda/lib/python3.10/site-packages/cudf/pandas/__init__.py, line 44, in install (ModuleAccelerator=<ABCMeta(__module__='cudf.pandas.module_accelerator', __annotations__={'_denylist': 'tuple[str]', '_use_fast_lib': 'bool', '_use_fast_lib_lock': 'threading.RLock', '_module_cache_prefix': 'str'}, __doc__='\n    A finder and loader that produces "accelerated" modules.\n\n    When someone attempts to import the specified slow library with\n    this finder enabled, we intercept the import and deliver an\n    equivalent, accelerated, version of the module. This provides\n    attributes and modules that check if they are being used from\n    "within" the slow (or fast) library themselves. If this is the\n    case, the implementation is forwarded to the actual slow library\n    implementation, otherwise a proxy implementation is used (which\n    attempts to call the fast version first).\n    ', _module_cache_prefix='_slow_lib_', __new__=<staticmethod at remote 0x7fb509a64a60>, _popu...(truncated), throwflag=<optimized out>)
    at /usr/local/src/conda/python-3.10.14/Python/ceval.c:4231
#12 0x0000561c3c2eea2c in _PyEval_EvalFrame (throwflag=0, 
    f=Frame 0x7fb6b3d9e5c0, for file /opt/conda/lib/python3.10/site-packages/cudf/pandas/__init__.py, line 44, in install (ModuleAccelerator=<ABCMeta(__module__='cudf.pandas.module_accelerator', __annotations__={'_denylist': 'tuple[str]', '_use_fast_lib': 'bool', '_use_fast_lib_lock': 'threading.RLock', '_module_cache_prefix': 'str'}, __doc__='\n    A finder and loader that produces "accelerated" modules.\n\n    When someone attempts to import the specified slow library with\n    this finder enabled, we intercept the import and deliver an\n    equivalent, accelerated, version of the module. This provides\n    attributes and modules that check if they are being used from\n    "within" the slow (or fast) library themselves. If this is the\n    case, the implementation is forwarded to the actual slow library\n    implementation, otherwise a proxy implementation is used (which\n    attempts to call the fast version first).\n    ', _module_cache_prefix='_slow_lib_', __new__=<staticmethod at remote 0x7fb509a64a60>, _popu...(truncated), tstate=0x561c3c5b30a0)
    at /usr/local/src/conda/python-3.10.14/Include/internal/pycore_ceval.h:46
#13 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=<optimized out>, locals=0x0, con=0x7fb5d6ee67b0, tstate=0x561c3c5b30a0)
    at /usr/local/src/conda/python-3.10.14/Python/ceval.c:5067
#14 _PyFunction_Vectorcall (func=<function at remote 0x7fb5d6ee67a0>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>)
    at /usr/local/src/conda/python-3.10.14/Objects/call.c:342
#15 0x0000561c3c2e3142 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x7fb6b3d99bb0, callable=<function at remote 0x7fb5d6ee67a0>, tstate=0x561c3c5b30a0)
--Type <RET> for more, q to quit, c to continue without paging--
    at /usr/local/src/conda/python-3.10.14/Include/cpython/abstract.h:114
#16 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x7fb6b3d99bb0, callable=<function at remote 0x7fb5d6ee67a0>)
    at /usr/local/src/conda/python-3.10.14/Include/cpython/abstract.h:123
#17 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7ffcbe3e2dc0, tstate=<optimized out>)
    at /usr/local/src/conda/python-3.10.14/Python/ceval.c:5893
#18 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=Frame 0x7fb6b3d99a40, for file /opt/work/./test2.py, line 2, in <module> (), throwflag=<optimized out>)
    at /usr/local/src/conda/python-3.10.14/Python/ceval.c:4181
#19 0x0000561c3c381c60 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x7fb6b3d99a40, for file /opt/work/./test2.py, line 2, in <module> (), tstate=0x561c3c5b30a0)
    at /usr/local/src/conda/python-3.10.14/Include/internal/pycore_ceval.h:46
#20 _PyEval_Vector (tstate=tstate@entry=0x561c3c5b30a0, con=con@entry=0x7ffcbe3e2ec0, 
    locals=locals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}, args=args@entry=0x0, argcount=argcount@entry=0, kwnames=kwnames@entry=0x0) at /usr/local/src/conda/python-3.10.14/Python/ceval.c:5067
#21 0x0000561c3c381ba7 in PyEval_EvalCode (co=co@entry=<code at remote 0x7fb6b3cab310>, 
    globals=globals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}, 
    locals=locals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}) at /usr/local/src/conda/python-3.10.14/Python/ceval.c:1134
#22 0x0000561c3c3b212a in run_eval_code_obj (tstate=tstate@entry=0x561c3c5b30a0, co=co@entry=0x7fb6b3cab310, 
--Type <RET> for more, q to quit, c to continue without paging--
    globals=globals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}, 
    locals=locals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}) at /usr/local/src/conda/python-3.10.14/Python/pythonrun.c:1291
#23 0x0000561c3c3ad523 in run_mod (mod=mod@entry=0x561c3c5f6860, filename=filename@entry='/opt/work/./test2.py', 
    globals=globals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}, 
    locals=locals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}, flags=flags@entry=0x7ffcbe3e30a8, arena=arena@entry=0x7fb6b3d73b90) at /usr/local/src/conda/python-3.10.14/Python/pythonrun.c:1312
#24 0x0000561c3c2446f5 in pyrun_file (fp=fp@entry=0x561c3c596450, filename=filename@entry='/opt/work/./test2.py', start=start@entry=257, 
    globals=globals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}, 
    locals=locals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/opt/work/./test2.py') at remote 0x7fb6b3c90e20>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7fb6b3e08900>, '__file__': '/opt/work/./test2.py', '__cached__': None, 'cudf': <module at remote 0x7fb6b3cfdc10>}, closeit=closeit@entry=1, flags=0x7ffcbe3e30a8) at /usr/local/src/conda/python-3.10.14/Python/pythonrun.c:1208
#25 0x0000561c3c3a79fe in _PyRun_SimpleFileObject (fp=0x561c3c596450, filename='/opt/work/./test2.py', closeit=1, flags=0x7ffcbe3e30a8)
--Type <RET> for more, q to quit, c to continue without paging--
    at /usr/local/src/conda/python-3.10.14/Python/pythonrun.c:456
#26 0x0000561c3c3a7594 in _PyRun_AnyFileObject (fp=0x561c3c596450, filename=filename@entry='/opt/work/./test2.py', closeit=closeit@entry=1, flags=flags@entry=0x7ffcbe3e30a8)
    at /usr/local/src/conda/python-3.10.14/Python/pythonrun.c:90
#27 0x0000561c3c3a478b in pymain_run_file_obj (skip_source_first_line=0, filename='/opt/work/./test2.py', program_name='/opt/conda/bin/python')
    at /usr/local/src/conda/python-3.10.14/Modules/main.c:357
#28 pymain_run_file (config=0x561c3c597650) at /usr/local/src/conda/python-3.10.14/Modules/main.c:376
#29 pymain_run_python (exitcode=0x7ffcbe3e30a4) at /usr/local/src/conda/python-3.10.14/Modules/main.c:591
#30 Py_RunMain () at /usr/local/src/conda/python-3.10.14/Modules/main.c:670
#31 0x0000561c3c3751f7 in Py_BytesMain (argc=<optimized out>, argv=<optimized out>) at /usr/local/src/conda/python-3.10.14/Modules/main.c:1090
#32 0x00007fb6b3f01d90 in ?? () from /usr/lib/x86_64-linux-gnu/libc.so.6
#33 0x00007fb6b3f01e40 in __libc_start_main () from /usr/lib/x86_64-linux-gnu/libc.so.6
#34 0x0000561c3c3750f1 in _start ()
```

</details>

<details><summary>backtrace from 'py-bt' (click me)</summary>

```text
Traceback (most recent call first):
  File "/opt/conda/lib/python3.10/site-packages/cudf/pandas/__init__.py", line 44, in install
    rmm.mr.PoolMemoryResource(
  File "/opt/work/./test2.py", line 2, in <module>
    cudf.pandas.install()
```

</details>

Reply:
Looking at just a biiiit more traceback, if I'm reading this correctly it looks like we're getting a mix of `spdlog` symbols coming from `nvcomp` and from `rmm` ?

```text
#0  0x00007fb5e0c51365 in void spdlog::pattern_formatter::handle_flag_<spdlog::details::scoped_padder>(char, spdlog::details::padding_info) ()
   from /opt/conda/lib/python3.10/site-packages/libcudf/lib64/libnvcomp.so
#1  0x00007fb5d6a61f59 in spdlog::pattern_formatter::compile_pattern_(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
   from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
#2  0x00007fb5d6a62c39 in spdlog::logger::set_pattern(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, spdlog::pattern_time_type) ()
   from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
```

They both have `spdlog::pattern_formatter` symbols defined as weak symbols.

```shell
spdlog-pattern-syms(){
    nm -gDC "${1}" \
    | grep -E 'spdlog\:\:pattern_formatter'
}

spdlog-pattern-syms \
    /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
```

<details><summary>symbol table (click me)</summary>

```text
0000000000081ee0 W void spdlog::pattern_formatter::handle_flag_<spdlog::details::scoped_padder>(char, spdlog::details::padding_info)
00000000000810c0 W void spdlog::pattern_formatter::handle_flag_<spdlog::details::null_scoped_padder>(char, spdlog::details::padding_info)
0000000000082d00 W spdlog::pattern_formatter::compile_pattern_(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
0000000000070830 W spdlog::pattern_formatter::format(spdlog::details::log_msg const&, fmt::v10::basic_memory_buffer<char, 250ul, std::allocator<char> >&)
000000000006f2b0 W spdlog::pattern_formatter::~pattern_formatter()
000000000006f150 W spdlog::pattern_formatter::~pattern_formatter()
000000000006f150 W spdlog::pattern_formatter::~pattern_formatter()
0000000000085380 W spdlog::pattern_formatter::clone() const
000000000007ec80 W std::unique_ptr<spdlog::pattern_formatter, std::default_delete<spdlog::pattern_formatter> >::~unique_ptr()
000000000007ec80 W std::unique_ptr<spdlog::pattern_formatter, std::default_delete<spdlog::pattern_formatter> >::~unique_ptr()
0000000000167740 V typeinfo for spdlog::pattern_formatter
000000000012b330 V typeinfo name for spdlog::pattern_formatter
0000000000168628 V vtable for spdlog::pattern_formatter
```

</details>

```shell
spdlog-pattern-syms \
    /opt/conda/lib/python3.10/site-packages/libcudf/lib64/libnvcomp.so
```

<details><summary>symbol table (click me)</summary>

```text
0000000000097330 W void spdlog::pattern_formatter::handle_flag_<spdlog::details::scoped_padder>(char, spdlog::details::padding_info)
0000000000098150 W void spdlog::pattern_formatter::handle_flag_<spdlog::details::null_scoped_padder>(char, spdlog::details::padding_info)
0000000000098f70 W spdlog::pattern_formatter::compile_pattern_(std::string const&)
00000000000917d0 W spdlog::pattern_formatter::format(spdlog::details::log_msg const&, fmt::v9::basic_memory_buffer<char, 250ul, std::allocator<char> >&)
000000000008bf40 W spdlog::pattern_formatter::~pattern_formatter()
000000000008dee0 W spdlog::pattern_formatter::~pattern_formatter()
000000000008dee0 W spdlog::pattern_formatter::~pattern_formatter()
000000000009a420 W spdlog::pattern_formatter::clone() const
00000000010c0718 V typeinfo for spdlog::pattern_formatter
0000000000157f90 V typeinfo name for spdlog::pattern_formatter
00000000010c1098 V vtable for spdlog::pattern_formatter
```

</details>

Reply:
Here on 24.10, it seems that this is still suffering from symbol-visibility issues.

I repeated the same steps as in the comments above, and see similar failures + mix of `spdlog::` symbols from `rmm` and `nvcomp`.

```text
0x00007ff7c8674365 in void spdlog::pattern_formatter::handle_flag_<spdlog::details::scoped_padder>(char, spdlog::details::padding_info) ()
   from /opt/conda/lib/python3.10/site-packages/libcudf/lib64/libnvcomp.so
(gdb) bt
#0  0x00007ff7c8674365 in void spdlog::pattern_formatter::handle_flag_<spdlog::details::scoped_padder>(char, spdlog::details::padding_info) ()
   from /opt/conda/lib/python3.10/site-packages/libcudf/lib64/libnvcomp.so
#1  0x00007ff7be5685f9 in spdlog::pattern_formatter::compile_pattern_(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
   from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
#2  0x00007ff7be56930b in spdlog::logger::set_pattern(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, spdlog::pattern_time_type) ()
   from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
#3  0x00007ff7be588f62 in rmm::mr::pool_memory_resource<rmm::mr::device_memory_resource>::try_to_expand(unsigned long, unsigned long, rmm::cuda_stream_view) ()
   from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
#4  0x00007ff7be54eb38 in ?? () from /opt/conda/lib/python3.10/site-packages/rmm/_lib/memory_resource.cpython-310-x86_64-linux-gnu.so
```

@robertmaynard @KyleFromNVIDIA do you think we need changes to `rmm` to change the visibility of those `spdlog` symbols?

Reply:
Yes we need to change the default symbol visibiliy for the librmm cython modules.

Reply:
> change the default visibility for the librmm cython modules

I've opened https://github.com/rapidsai/rmm/pull/1644 with a proposed implementation of that.

Pushed some commits to this PR to install wheels from that PR's CI, and saw that `unit-test-cudf-pandas` job now succeeds!

build link: https://github.com/rapidsai/cudf/actions/runs/10323398706/job/28581417972?pr=15483

Reply:
I pushed some changes to https://github.com/rapidsai/rmm/pull/1644, described in https://github.com/rapidsai/rmm/pull/1644#issuecomment-2285079979.

Updated this PR to install wheels from that PR's CI, and still saw all the wheel test jobs succeed!

build link: https://github.com/rapidsai/cudf/actions/runs/10371174400/job/28712006548?pr=15483

So I *think* those `rmm` changes would fix the symbol-visibility issues we saw here.

Reply:
https://github.com/rapidsai/rmm/pull/1644 did fix this, so I think this thread can be resolved.

And I don't think we need to wait on https://github.com/rapidsai/rmm/pull/1654 to be merged, because https://github.com/rapidsai/rmm/pull/1644 added CI checks in `rmm` to check the type of conflicts that were causing issues here.